### PR TITLE
[#100] Config for association indexing

### LINKED ISSUE #100: Index associations
The current method of indexing associations has been superseded by trust networks. Instead it should be possible to index all associations similarly to how anchors are indexed. No specific meaning is given to the association type, it's stored as-is.

Similar to anchors, it should be possible to configure to index none, trust, or all.

### PR DESCRIPTION
**Pull request by brunodb3**
- Adds configuration for choosing which association transactions to index
  - `none`, `trust` or `all`
    - If `none`, will not index any associations
    - If `trust`, will only index associations from addresses in the trust network
    - If `all`, will index all associations
- Removes distinction of types for association transactions
  - No specific meaning is given to association types, it's stored as-is
- Removes association types and root address configuration

---

closes #100

### CODE REVIEW

File: src/associations/associations.service.ts [hunk 1]
```diff
@@ -19,56 +18,35 @@ export class AssociationsService implements OnModuleInit {
...
-  async createAssocIndex(transaction: Transaction) {
 
-    const {sender, associationType} = transaction;
+    const associationIndexing = this.config.getAssociationIndexing();
+    const senderRoles = await this.storage.getRolesFor(sender);
+    const isSenderTrustNetwork = Object.keys(senderRoles).length > 0;
 
-    if (this.config.getAssociationTypes().indexOf(associationType) === -1) {
-      this.logger.debug(`Ingoring association because unknown type ${associationType}`);
+    if (associationIndexing === 'none') {
```
Comment (jasny) [inline src/associations/associations.service.ts:34]: If associationIndexing is set to 'none', don't enable the listener.

File: src/associations/associations.service.ts [hunk 2]
```diff
@@ -19,56 +18,35 @@ export class AssociationsService implements OnModuleInit {
...
-      return this.removeAssocIndex(transaction);
+      this.logger.debug(`association-service: Unknown transaction type`);
+      return;
     }
-  }
-
-  async createAssocIndex(transaction: Transaction) {
 
-    const {sender, associationType} = transaction;
+    const associationIndexing = this.config.getAssociationIndexing();
```
Comment (jasny) [inline src/associations/associations.service.ts:30]: Get this in the listener and pass it to `index()` as argument.

File: src/associations/associations.service.ts [hunk 3]
```diff
@@ -19,56 +18,35 @@ export class AssociationsService implements OnModuleInit {
...
+    const senderRoles = await this.storage.getRolesFor(sender);
+    const isSenderTrustNetwork = Object.keys(senderRoles).length > 0;
```
Comment (jasny) [inline src/associations/associations.service.ts:32]: Move this into the `if`, so this is only executed if indexing is set to 'trust'.

File: src/config/data/default.schema.json [hunk 4]
```diff
@@ -47,6 +25,13 @@
     },
     "env": "TRUST_NETWORK_ROLES"
   },
+  "association": {
+    "indexing": {
+      "default": "all",
```
Comment (jasny) [inline src/config/data/default.schema.json:30]: default to 'none'

**Review by jasny — CHANGES_REQUESTED** [review]

File: src/associations/associations.service.ts (hunk 1 above)
Comment (brunodb3) [reply src/associations/associations.service.ts]: done

File: src/associations/associations.service.ts (hunk 2 above)
Comment (brunodb3) [reply src/associations/associations.service.ts]: done

File: src/associations/associations.service.ts (hunk 3 above)
Comment (brunodb3) [reply src/associations/associations.service.ts]: done

File: src/config/data/default.schema.json (hunk 4 above)
Comment (brunodb3) [reply src/config/data/default.schema.json]: done

**Review by jasny — APPROVED** [review]

**Pull request merged**